src/core/qgsfileutils.cpp

```diff
@@ -228,3 +244,41 @@ QStringList QgsFileUtils::findFile( const QString &file, const QString &basePath
   return foundFiles;
 }
 
+
+
+QgsFileSearchTask::QgsFileSearchTask( const QString &file, const QString &basePath, int maxClimbs, int searchCeilling, const QString &currentDir, QProgressDialog *dialog, QStringList *container  )
+  : QgsTask( tr( "searching for %1" ).arg( file ), QgsTask::CanCancel )
+  , mFile( file )
+  , mBasePath( basePath )
+  , mMaxClimbs( maxClimbs )
+  , mSearchCeil( searchCeilling )
+  , mCurrentDir( currentDir )
+  , mDialog( dialog )
+  , mResults( container )
+{
+  connect( dialog, QProgessDialog::canceled, this, QgsFileSearchTask::cancel );
```

Comment:
You can drop the dialog handling here -- QgsTask /QgsTaskManager does its own progress reporting (and it's not safe to access any gui component like a dialog from a background thread, this crashes Qt)

If you want to show the progress in a specific dialog outside of the standard task manager widget, you'd connect the task's progressChanged signal to your dialog's setProgress slot (and similarly with the cancelation -- connect the dialog's canceled signal to the tasks' cancel slot), but setup this in the main thread only.

Reply:
Since using process events was clunky, I want to be able to provide a way to stop the whole process from the gui. Not sure what is the best option and if the task manager will only stop one lookup and not the whole process.